packages/clarity-js/src/core/report.ts

```diff
@@ -8,15 +9,18 @@ export function reset(): void {
     history = [];
 }
 
-export function report(message: string): void {
+export function report(check: Check, message: string = null): void {
     // Do not report the same message twice for the same page
     if (history && history.indexOf(message) === -1) {
         const url = config.report;
         if (url && url.length > 0) {
-            let payload = JSON.stringify({m: message, p: data.projectId, u: data.userId, s: data.sessionId, n: data.pageNum} as Report);
+            let payload: Report = {c: check, p: data.projectId, u: data.userId, s: data.sessionId, n: data.pageNum };
+            if (message) payload.m = message;
+            // Using POST request instead of a GET request (img-src) to not violate existing CSP rules
```

Comment:
>instead [](http://example.com/codeflow?start=34&length=7)

Can you explain how this will not violate existing CSP rules and why this is better for this?How do we avoid the reporting endpoint itself getting CSP issues?

Reply:
This was mostly a note for future reference. There is no way to ensure report end-point will comply with CSP rules because third party sites can choose to have their own CSP rules. However, what this does is that it ensures that report end-point stays within the same boundaries that are necessary for core functionality of Clarity. 

More concretely, if a site intentionally adds Clarity from say xyz.com domain, they will likely also add following CSP rule: connect-src: xyz.com (to allow for data to be uploaded to xyz.com using XHR). Which is what report end-point uses right now. If we instead use an image to create a GET request then connect-src permissions will no longer be sufficient and the site will also need to add img-src: xyz.com to CSP before report end-point can work.